CMake uninstall target conflicts with other previously registered uninstall targets

I am trying to build netcdf-c & netcdf-cxx4 within a cmake project via [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html). Both netcdf-c and netcdf-cxx4 register an `uninstall` which violates [CMP0002](https://cmake.org/cmake/help/latest/policy/CMP0002.html) and causes cmake to fail with this error:

```
CMake Error at build/_deps/netcdf-cxx4-src/CMakeInstallation.cmake:109 (ADD_CUSTOM_TARGET):
  ADD_CUSTOM_TARGET cannot create target "uninstall" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "build/_deps/netcdf-c-src".
  See documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  build/_deps/netcdf-cxx4-src/CMakeLists.txt:688 (INCLUDE)
```

Other projects [have solved this](https://gitlab.com/libeigen/eigen/-/issues/1842) issue by wrapping the custom target in an `option()` so that users can disable the target.

This is the same issue as https://github.com/Unidata/netcdf-c/issues/2597

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@
 ##################################
 
 #Minimum required CMake Version
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.21.0)
 
 #Project Name
 PROJECT(NCXX C CXX)
@@ -685,4 +685,6 @@ ENDIF()
 ##
 # CPack, CMakeInstallation.cmake file.
 ##
-INCLUDE(CMakeInstallation.cmake)
+if (PROJECT_IS_TOP_LEVEL)
+  include(CMakeInstallation.cmake)
+endif()

